consensus/src/event_processor.rs

```diff
@@ -396,63 +364,48 @@ impl<T: Payload> EventProcessor<T> {
         author: Author,
         help_remote: bool,
     ) -> anyhow::Result<()> {
-        if help_remote {
-            self.help_remote_if_stale(author, sync_info.highest_round(), sync_info.hqc_round());
-        }
-
-        let current_hqc_round = self
-            .block_store
-            .highest_quorum_cert()
-            .certified_block()
-            .round();
-
-        let current_htc_round = self
-            .block_store
-            .highest_timeout_cert()
-            .map_or(0, |tc| tc.round());
-
-        if current_hqc_round >= sync_info.hqc_round()
-            && current_htc_round >= sync_info.htc_round()
-            && self.block_store.root().round()
-                >= sync_info.highest_commit_cert().commit_info().round()
-        {
+        let local_sync_info = self.gen_sync_info();
+        if sync_info.is_stale(&local_sync_info) {
+            if help_remote {
+                counters::SYNC_INFO_MSGS_SENT_COUNT.inc();
+                debug!(
+                    "Peer {} has stale state {}, send it back {}",
+                    author.short_str(),
+                    sync_info,
+                    local_sync_info,
+                );
+                self.network.send_sync_info(local_sync_info, author);
+            }
             return Ok(());
         }
-
-        // Some information in SyncInfo is ahead of what we have locally.
-        // First verify the SyncInfo (didn't verify it in the yet).
-        sync_info.verify(&self.epoch_info().verifier).map_err(|e| {
-            security_log(SecurityEvent::InvalidSyncInfoMsg)
-                .error(&e)
-                .data(&sync_info)
-                .log();
-            e
-        })?;
-
-        debug!(
-            "Starting sync: current_hqc_round = {}, sync_info_hqc_round = {}",
-            current_hqc_round,
-            sync_info.hqc_round(),
-        );
-        self.block_store
-            .sync_to(&sync_info, self.create_block_retriever(author))
-            .await
-            .map_err(|e| {
-                warn!(
-                    "Fail to sync up to HQC @ round {}: {}",
-                    sync_info.hqc_round(),
-                    e
-                );
+        if local_sync_info.is_stale(&sync_info) {
+            debug!(
+                "Local state {} is stale than peer {} remote state {}",
+                local_sync_info,
+                author.short_str(),
+                sync_info
+            );
+            // Some information in SyncInfo is ahead of what we have locally.
+            // First verify the SyncInfo (didn't verify it in the yet).
+            sync_info.verify(&self.epoch_info().verifier).map_err(|e| {
+                security_log(SecurityEvent::InvalidSyncInfoMsg)
```

Comment:
we should probably start stripping out security logs and placing the right libra log / structured log here...

Reply:
yeah, a lot of cleanup work to do